yt/yt/ytlib/api/native/client_authentication_impl.cpp

```diff
@@ -243,18 +261,29 @@ TListUserTokensResult TClient::DoListUserTokens(
     }
 
     std::vector<TString> userTokens;
+    std::map<TString, NYson::TYsonString> tokensMetadata;
 
     auto tokens = ConvertTo<IListNodePtr>(rspOrError.Value());
     for (const auto& tokenNode : tokens->GetChildren()) {
         const auto& attributes = tokenNode->Attributes();
         auto userAttribute = attributes.Find<TString>("user");
         if (userAttribute == user) {
             userTokens.push_back(ConvertTo<TString>(tokenNode));
+            if (options.WithMetadata) {
+                auto metadata = BuildYsonStringFluently()
+                    .BeginMap()
+                        .Item("description").Value(attributes.Find<TString>("description"))
+                        .Item("creation_time").Value(attributes.Find<TString>("creation_time"))
+                        .Item("effective_expiration").Value(attributes.GetYson("effective_expiration"))
```

Comment:
What is the type of effective_expiration here? Isn't it also TString?

Reply:
https://ytsaurus.tech/docs/ru/user-guide/storage/cypress#TTL

It is actually a map of 2 elements:
```
{
  "time": {"value": 42, "path": //testator/path}
  "timeout": {"value": 42, "path": //testator/path}
}
```

time represents absolute time value of expiration
timeout represents amount of inactivity(no access to the element) after which the element will be deleted

Currently we don't have a public way to set `time` (only cluster admins can do it manually for specific tokens or for `//sys/cypress_tokens`).

YQL agents create temporary tokens with `timeout` and refresh them, so user will see them.

However, using a token doesn't refresh the timeout(doesn't count as an "access") currently, but may be changed in the future

Reply:
Ok. I see